src/Symfony/Component/Routing/RouteCollectionBuilder.php

```diff
@@ -238,6 +240,20 @@ public function setMethods($methods)
     }
 
     /**
+     * Adds a resource for this collection.
+     *
+     * @param ResourceInterface $resource
+     *
+     * @return $this
+     */
+    private function addResource(ResourceInterface $resource)
```

Comment:
It *can* be private. As odd as it sees, since this method is being called from inside `RouteCollectionBuilder` (even though it's not being called on `$this`, but another object), it is legal to call a private method. This is done in Silex in several places. However, I would be quite happy if this were public. I think (but don't really know) that people may eventually ask for this method if it's not present.